packages/components/src/styles/mixins/_interactive-dark-theme.scss

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+ @use "../mixins/focus-ring" as *;
+ @use "../mixins/button" as *;
+
+ // TODO: Deprecate SideNav specific dark theme tokens in favor of general dark theme tokens
+
+// Used to apply dark theme to interactive elements such as Button & Dropdown
+@mixin hds-interactive-dark-theme($add-visible-border: true, $is-within-side-nav: false) {
+  @if $is-within-side-nav {
+    // sidenav specific dark theme tokens
+    color: var(--token-side-nav-color-foreground-strong);
+    background-color: transparent; // test
+    border-radius: var(--token-side-nav-body-list-item-border-radius);
+  } @else {
+    // general dark theme tokens
+    color: var(--token-color-palette-neutral-0);
+    background-color: var(--token-color-palette-neutral-700);
+    border-radius: $hds-button-border-radius; // We don't have a button border-radius token so we use the button mixin sass variable
+  }
+  
+  border: 1px solid transparent;
+  cursor: pointer;
+
+  @if ($add-visible-border) {
+    border-color: var(--token-color-palette-neutral-500);
+  }
+
+  &:focus,
+  &:focus-visible,
+  &.mock-focus {
+    @include hds-focus-ring-with-pseudo-element($top: -1px, $right: -1px, $bottom: -1px, $left: -1px);
+
+    color: var(--token-color-palette-neutral-0);
+    background-color: var(--token-color-palette-neutral-700);
+
+    @if ($add-visible-border) {
+      border-color: var(--token-color-palette-neutral-500); // TODO: confirm if this is the correct token
```

Comment:
@jorytindall Is this the correct border color for the focused Buttons/Dropdowns?

Reply:
I've better mapped these out [here](https://www.figma.com/design/noyY6dUMDYjmySpHcMjhkN/branch/5RLNe1engBID42LhAm75NN/HDS-Components?m=auto&node-id=59659-45808&t=CICqCkO7FCafQBOb-1), but the intention is that it should match the re-themed dropdown.

Reply:
Pushed some updates to better match mocks